roles/config-pip-packages/tasks/main.yml

```diff
@@ -0,0 +1,19 @@
+---
+
+- name: Install Python software packages
+  pip:
+    name: "{{ item }}"
+  loop: "{{ pip.packages | default([]) }}"
+  when:
+    - pip is defined
+    - pip.virtualenv is not defined
+
+- name: Install Python software packages using {{ pip.virtualenv }}
+  pip:
+    name: "{{ item }}"
+    virtualenv: "{{ pip.virtualenv }}"
```

Comment:
This task is very much the same as the one above with the exception of the `virtualenv` vars ... In those cases I'd recommend using the `omit` keyword to avoid duplicating tasks (and doing a lot of checks) to keep it simple yet flexible. For example:

```
 : 
  pip:
    name: "{{ item }}"
    virtualenv: "{{ pip.virtualenv | default(omit) }}"
```

Reply:
TIL about `omit` :+1: 
Thank you @oybed, see update.